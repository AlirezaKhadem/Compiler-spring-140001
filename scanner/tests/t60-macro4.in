import "define.d"
define DEFINE "DEFINE"
define DeFiNe 2+ 3
define x singleX
define xx doubleX
define xxx tripleX

__func__
func
"func"
__line__
line
DEFINE "define"
Print(DeFiNe)

x[2]
xxxx[2]
xx
(x)
a xxx b
axb
axxb
axxxb
abx
xba
anString = "xyzxxtxxx"
anotherString = "xx"
