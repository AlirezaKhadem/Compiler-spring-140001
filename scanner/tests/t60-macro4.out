import
T_STRINGLITERAL "define.d"
__func__
T_ID func
T_STRINGLITERAL "func"
__line__
T_ID line
T_STRINGLITERAL "DEFINE"
T_STRINGLITERAL "define"
Print
(
T_INTLITERAL 2
+
T_INTLITERAL 3
)
T_ID singleX
[
T_INTLITERAL 2
]
T_ID xxxx
[
T_INTLITERAL 2
]
T_ID doubleX
(
T_ID singleX
)
T_ID a
T_ID tripleX
T_ID b
T_ID axb
T_ID axxb
T_ID axxxb
T_ID abx
T_ID xba
T_ID anString
=
T_STRINGLITERAL "xyzxxtxxx"
T_ID anotherString
=
T_STRINGLITERAL "xx"
